SleepEngine/Engine/Utils/stl_utils.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#pragma region Check Find
+
+template <class TContainer>
+bool Contains(TContainer const& container, typename TContainer::const_reference value)
+{
+    return std::find(container.cbegin(), container.cend(), value) != container.cend();
+}
+
+template <class TContainer, class TPred>
+bool ContainsIf(TContainer const& container, TPred pred)
+{
+    return FindIf(container, pred) != container.cend();
+}
+
+template <class TContainer, class TPred>
+bool All(TContainer const& container, TPred pred)
+{
+    return std::all_of(container.cbegin(), container.cend(), pred);
+}
+
+// true if no elements from v1 are found in v2
+template <class T>
+bool AreDistinct(std::vector<T> const& v1, std::vector<T> const& v2)
+{
+    return Intersection(v1, v2).empty();
+}
+
+// complexity: len v1 * len v2
+template <class TContainer, class TOtherContainer>
+bool AreDistinctFind(TContainer const& v1, TOtherContainer const& v2)
+{
+    using Ref = typename TContainer::const_reference;
+    static_assert(std::is_same_v<Ref, typename TOtherContainer::const_reference>,
+        "TContainer and TOtherContainer must hold elements of same type");
+
+    return All(v1, [&v2](Ref element){
+            return !Contains(v2, element);
+        });
+}
+
+template <class TContainer, class TPred>
+typename TContainer::const_iterator FindIf(TContainer const& container, TPred pred)
+{
+    return std::find_if(container.cbegin(), container.cend(), pred);
+}
+
+template <class TContainer>
+bool AreUnique(TContainer container)
+{
+    auto oldEnd = container.cend();
+    auto it = std::unique(container.begin(), container.end());
+    return it == oldEnd;
+}
```

Comment:
this is so bad that you copy container here. I am sure there are better solutions for implementation, without copying

Reply:
where? i don't see any copying above...

Reply:
You must have meant the intersection function. I must do it, because sorting of the arguments is in place